libutils/alloc.h

```diff
@@ -39,6 +39,14 @@ void *xmemdup(const void *mem, size_t size);
 int xasprintf(char **strp, const char *fmt, ...) FUNC_ATTR_PRINTF(2, 3);
 int xvasprintf(char **strp, const char *fmt, va_list ap) FUNC_ATTR_PRINTF(2, 0);
 
+static inline void free_array_items(void **array, size_t n_items)
+{
+    for (size_t i = 0; i < n_items; i++)
```

Comment:
should add an if or assert for `array == NULL`. The norm is that this should be a no-op, but you might have a good reason for why it should not be allowed.

Reply:
It's not free'ing the array itself. So an `assert()` it is. :+1: